src/webgpu/shader/execution/builtin/inversesqrt.spec.ts

```diff
@@ -0,0 +1,111 @@
+export const description = `
+Execution Tests for the 'inverseSqrt' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { f32, f32Bits, Scalar, TypeF32, u32 } from '../../../util/conversion.js';
+
+import { Case, Config, kBit, run, ulpThreshold } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+// Calculates the linear interpolation between two values of a given ratio
+function lerp(min: number, max: number, t: number) {
+  if (!Number.isFinite(min) || !Number.isFinite(max) || !Number.isFinite(t)) {
+    return Number.NaN;
+  }
+  if (min === max) {
+    return min;
+  }
+
+  if (min > max) {
+    const temp = min;
+    min = max;
+    max = temp;
+  }
+
+  if (t > 1) {
+    return max;
+  }
+
+  if (t < 0) {
+    return min;
+  }
+
+  return min + t * (max - min);
+}
+
+// Unwrap Scalar params into numbers and test preconditions
+function unwrapRangeParams(min: Scalar, max: Scalar, steps: Scalar) {
+  assert(min.type.kind === 'f32', '|min| needs to be a f32');
+  assert(max.type.kind === 'f32', '|max| needs to be a f32');
+  assert(steps.type.kind === 'u32', '|steps| needs to be a u32');
+
+  const f32_min = min.value as number;
+  const f32_max = max.value as number;
+  const u32_steps = steps.value as number;
+
+  assert(f32_max > f32_min, '|max| must be greater than |min|');
+  assert(u32_steps > 0, '|steps| must be greater than 0');
+
+  return { f32_min, f32_max, u32_steps };
+}
+
+// Linear range generator, since TypeScript doesn't provide one
+function linearRange(min: Scalar, max: Scalar, steps: Scalar): Array<number> {
+  const { f32_min, f32_max, u32_steps } = unwrapRangeParams(min, max, steps);
+
+  return Array.from(Array(u32_steps).keys()).map(i => lerp(f32_min, f32_max, i / (u32_steps - 1)));
+}
+
+// Non-linear range generator, biased to towards min.
+function biasedRange(min: Scalar, max: Scalar, steps: Scalar): Array<number> {
+  const { f32_min, f32_max, u32_steps } = unwrapRangeParams(min, max, steps);
+
+  return Array.from(Array(u32_steps).keys()).map(i =>
+    lerp(f32_min, f32_max, Math.pow(lerp(0, 1, i / (u32_steps - 1)), 2))
+  );
+}
```

Comment:
All of the range generation code can probably be factored out into a utility file and used by other tests that test a range of programmatically values.

I wanted to get this reviewed before I did that refactoring, and also wasn't sure if I should do that in a separate PR. i.e. land this test PR and then have a follow up that factors out the utility code and update other tests to use it.

Reply:
Once this PR lands I will refactor and send out another PR.